app/models/validations/sales/sale_information_validations.rb

```diff
@@ -348,6 +356,20 @@ def validate_mortgage_used_dont_know(record)
     end
   end
 
+  def validate_number_of_staircase_transactions(record)
+    return unless record.numstair
+
+    if record.firststair == 2 && record.numstair < 2
+      record.errors.add :numstair, I18n.t("validations.sales.sale_information.numstair.must_be_greater_than_one")
+      record.errors.add :firststair, I18n.t("validations.sales.sale_information.firststair.cannot_be_no")
+    end
+
+    if record.firststair == 1 && record.numstair > 1
+      record.errors.add :numstair, I18n.t("validations.sales.sale_information.numstair.must_be_one")
+      record.errors.add :firststair, I18n.t("validations.sales.sale_information.firststair.cannot_be_yes")
+    end
```

Comment:
Is this needed? You can't navigate to the questions to enter the values like this but you could in a bulk upload.

Reply:
I think it would get inferred for BU as well, we do something similar with `beds` when it's a bedsit